middleware/proxy.go

```diff
@@ -72,6 +72,11 @@ type (
 		Next(echo.Context) *ProxyTarget
 	}
 
+	// TargetProvider defines an interface that gives the opportunity for balancer to return custom errors when selecting target.
+	TargetProvider interface {
+		GetTarget(echo.Context) (*ProxyTarget, error)
```

Comment:
If think `NextTarget` would be better as those other methods on `ProxyBalancer` interface also have `*Target` suffix and by having `Next*` prefix it should be clearer that it is some variation of `Next()`

Reply:
Should the name of the interface to be NextTarget as well?